cdap-api/src/main/java/co/cask/cdap/api/artifact/PluginConfigurer.java

```diff
@@ -25,7 +26,7 @@
  * This interface provides methods to register plugin usage in a CDAP Program. The registered plugins will be
  * available at the runtime of the Program.
  */
-public interface PluginConfigurer {
+public interface PluginConfigurer extends DatasetConfigurer {
```

Comment:
Sure, will allow Applications to add Plugins as well, which will be available to all programs. But currently not adding the ability to add plugins to Flows/Services. Need to figure out there is a use case for it.